### ISSUE TITLE
Issue: Cannot use CombinedMemory with VectorStoreRetrieverMemory and ConversationTokenBufferMemory

### ISSUE THREAD

**Issue by tnunamak** [langchain-ai/langchain#7695]
### Issue you'd like to raise.

Here is an (incomplete) modification of the documentation for using `CombinedMemory`, but with `VectorStoreRetrieverMemory` instead of `ConversationSummaryMemory`.

```python
from langchain.llms import OpenAI
from langchain.prompts import PromptTemplate
from langchain.chains import ConversationChain
from langchain.memory import (
    ConversationBufferMemory,
    CombinedMemory,
    ConversationSummaryMemory,
)


conv_memory = ConversationBufferMemory(
    memory_key="chat_history_lines", input_key="input"
)
# not shown: define retriever as shown in the [Chroma docs](https://python.langchain.com/docs/modules/data_connection/vectorstores/integrations/chroma)
vector_memory = VectorStoreRetrieverMemory(retriever=retriever, memory_key="history", input_key="input")
    
# Combined
memory = CombinedMemory(memories=[vector_memory, summary_memory])
_DEFAULT_TEMPLATE = """The following is a friendly conversation between a human and an AI. The AI is talkative and provides lots of specific details from its context. If the AI does not know the answer to a question, it truthfully says it does not know.

Summary of conversation:
{history}
Current conversation:
{chat_history_lines}
Human: {input}
AI:"""
PROMPT = PromptTemplate(
    input_variables=["history", "input", "chat_history_lines"],
    template=_DEFAULT_TEMPLATE,
)
llm = OpenAI(temperature=0)
conversation = ConversationChain(llm=llm, verbose=True, memory=memory, prompt=PROMPT)
```

What happens is, [on this line](https://github.com/hwchase17/langchain/blob/master/langchain/memory/vectorstore.py#L58), `inputs.items()` looks something like this:

```python
dict_items([('input', 'How are you?'), ('chat_history_lines', 'Human: wow\nAI: Wow!'), ('history', 'Human: wow\nAI: Wow!\nHuman:yes\nAI: Yes!\nHuman: hello\nAI: Hi!')])
```

When adding documents to the vectorstore retriever memory, all items are added except for `self.memory_key` (history). Thus, `chat_history_lines` is included in the created documents, without a way to prevent that.

### Suggestion:

One approach for supporting this would be to add a property to VectorStoreRetrieverMemory that allows the caller to specify which input keys should be included in the created documents. Another approach may be to only create documents with input_key from inputs.

**Comment by keenborder786** [comment]
@tnunamak I might be able to help but not sure what do you mean "When adding documents to the vectorstore retriever memory, all items are added except for self.memory_key (history). Thus, chat_history_lines is included in the created documents, without a way to prevent that."

**Comment by tnunamak** [comment]
Thanks @keenborder786. The use case is that I want to use combined memory that consists of two parts:

1. A Chroma vector store containing each input and response for the ConversationChain. With this, I can include something like this in my prompt: "Some relevant comments from the conversation include {history}", using `memory_db.as_retriever(search_type="mmr")`.
2. ConversationBufferMemory that contains the most recent input and response for the ConversationChain. With this, I can include something like this in my prompt: "The last few messages from the conversation are {chat_history_lines}".

Basically, my goal is for both relevant items in history (as constructed via mmr search) and the most recent items in history to be passed as inputs to my prompt, using CombinedMemory.

That said, for now, I can work around this issue by manually injecting `chat_history_lines` into the prompt via `memory.load_memory_variables({})["chat_history_lines"]`.

**Comment by keenborder786** [comment]
@tnunamak  something like will work right:
```
vector_memory = VectorStoreRetrieverMemory(retriever=retriever, memory_key="history", input_key="input",exclude_input_keys = ["chat_history_lines"])
```
The exclude_input_key will exclude the input generated by the key when constructing the document. This means the result of ConversationBufferMemory(memory_key = 'chat_history_lines') will be excluded when constructing the document for VectorStoreRetriver Memory.

**Comment by keenborder786** [comment]
Following is a more concrete usage example:

```python

embeddings = OpenAIEmbeddings()
db = Chroma.from_documents([Document(page_content = '', metadata = {})] , embeddings)
retriever = db.as_retriever(search_type="mmr")



conv_memory = ConversationBufferMemory(
    memory_key="chat_history_lines", input_key="input"
)
vector_memory = VectorStoreRetrieverMemory(retriever=retriever, memory_key="history", input_key="input",exclude_input_keys = ["chat_history_lines"])
    
# # Combined
memory = CombinedMemory(memories=[vector_memory, conv_memory])
_DEFAULT_TEMPLATE = """The following is a friendly conversation between a human and an AI. The AI is talkative and provides lots of specific details from its context. If the AI does not know the answer to a question, it truthfully says it does not know.

Summary of conversation:
{history}
Current conversation:
{chat_history_lines}
Human: {input}
AI:"""
PROMPT = PromptTemplate(
    input_variables=["history","chat_history_lines", "input"],
    template=_DEFAULT_TEMPLATE,
)

llm = OpenAI(temperature=0)
conversation = ConversationChain(llm=llm, verbose=True, memory=memory, prompt=PROMPT)
# print(conversation.validate_prompt_input_variables())
while True:
    print(conversation.predict(input = input()))


```

As you can the see in the output below that when using combined memory the history input of VectorStore excludes the history created by ConversationBufferMemory

```bash

The following is a friendly conversation between a human and an AI. The AI is talkative and provides lots of specific details from its context. If the AI does not know the answer to a question, it truthfully says it does not know.

Summary of conversation:
input: How aHow are you doing?
response:  I'm doing great! How about you?
input: Hello
response:  Hi there! How can I help you?

Current conversation:
Human: Hello
AI:  Hi there! How can I help you?
Human: How aHow are you doing?
AI:  I'm doing great! How about you?
Human: How is the weather?
AI:

```

**Comment by tnunamak** [comment]
Thanks, I didn't notice exclude_input_keys. Look forward to giving this a try, I'll report back!

**Comment by tnunamak** [comment]
That worked, thank you for taking the time and effort to break it down and walk me through it @keenborder786!

**Comment by tnunamak** [comment]
@keenborder786 I was mistaken, this doesn't seem to be working for me. I couldn't find any documentation about `exclude_input_keys`. Can I ask where you got it from?

When I tried running your example, here's what I got:

```
The following is a friendly conversation between a human and an AI. The AI is talkative and provides lots of specific details from its context. If the AI does not know the answer to a question, it truthfully says it does not know.

Summary of conversation:
input: How are you doing?
chat_history_lines: Human: Hello
AI:  Hi there! How can I help you?
response:  I'm doing great! I'm feeling very excited to be able to help you with whatever you need. What can I do for you?
input: Hello
chat_history_lines: 
response:  Hi there! How can I help you?

Current conversation:
Human: Hello
AI:  Hi there! How can I help you?
Human: How are you doing?
AI:  I'm doing great! I'm feeling very excited to be able to help you with whatever you need. What can I do for you?
Human: How is the weather?
AI:
```

Here is the full script I used

```python
from langchain.llms import OpenAI
from langchain.prompts import PromptTemplate
from langchain.chains import ConversationChain
from langchain.memory import (
    ConversationBufferMemory,
    CombinedMemory,
    ConversationSummaryMemory,
)

from langchain.vectorstores import Chroma
from langchain.embeddings.openai import OpenAIEmbeddings

from langchain.memory import (
    CombinedMemory,
    ConversationTokenBufferMemory,
    VectorStoreRetrieverMemory,
)
from langchain.docstore.document import Document


embeddings = OpenAIEmbeddings()
db = Chroma.from_documents([Document(page_content = '', metadata = {})] , embeddings)
retriever = db.as_retriever(search_type="mmr")



conv_memory = ConversationBufferMemory(
    memory_key="chat_history_lines", input_key="input"
)
vector_memory = VectorStoreRetrieverMemory(retriever=retriever, memory_key="history", input_key="input",exclude_input_keys = ["chat_history_lines"])

# # Combined
memory = CombinedMemory(memories=[vector_memory, conv_memory])
_DEFAULT_TEMPLATE = """The following is a friendly conversation between a human and an AI. The AI is talkative and provides lots of specific details from its context. If the AI does not know the answer to a question, it truthfully says it does not know.

Summary of conversation:
{history}
Current conversation:
{chat_history_lines}
Human: {input}
AI:"""
PROMPT = PromptTemplate(
    input_variables=["history","chat_history_lines", "input"],
    template=_DEFAULT_TEMPLATE,
)

llm = OpenAI(temperature=0)
conversation = ConversationChain(llm=llm, verbose=True, memory=memory, prompt=PROMPT)
# print(conversation.validate_prompt_input_variables())
while True:
    print(conversation.predict(input = input()))
    ```

**Comment by keenborder786** [comment]
@tnunamak I was also surprised that you closed the issue since `exclude_input_keys` is not a feature. I will implement this. I was just clarifying about your requirements (apologies for the confusion). I see that you need the same feature so I will be opening a PR and then you will be able pass `exclude_input_keys`.

**Comment by keenborder786** [comment]
@tnunamak done please see the PR
